src/util.ts

```diff
@@ -140,3 +153,84 @@ export function isYarnUsed(existsSync = fs.existsSync): boolean {
 export function getPkgManagerName(isYarnUsed?: boolean): 'yarn' | 'npm' {
   return isYarnUsed ? 'yarn' : 'npm';
 }
+
+/**
+ * Create a directory if not exist
+ * If already exist check if contains any ts file.
+ * If it returns true we can expect that the directory
+ * exist and does not contain ts files.
+ */
+export async function createSrcDir(
+  dirpath: string,
+  options?: Options
+): Promise<boolean> {
+  try {
+    fs.mkdirSync(dirpath, { mode: 0o774 });
```

Comment:
is there a reason to pass in the mode? This may end up being more permissive than what the user wanted. I suspect that the Node.js core official docs are incorrect - they claim the default to be `0o777`. That's not the actual behavior though:

![image](https://user-images.githubusercontent.com/79017/56514193-8602d180-64e9-11e9-9f9f-c9481b2471fb.png)

Reply:
That's right the official doc is out of sync, I just wanted to avoid the 0777 but leave the default behavior is better.